infra/blueprint-test/README.md

```diff
@@ -0,0 +1,409 @@
+
+```hcl
+// name for your example module
+module "mysql-db" { 
+ // set the source for the module being tested as part of the 
+ // example
+ source               = "../../modules/mysql"
+ ...
+}
+```
+
+4. Create a new file named `variables.tf` for organizing and defining variables that need to be passed into the example module as follows:
+
+```hcl
+...
+variable "project_id" {
+ description = "The ID of the project in which resources will be provisioned."
+ type        = string
+}
+ 
+variable "db_name" {
+ description = "The name of the SQL Database instance"
+ default     = "example-mysql-public"
+}
+...
+```
+
+These variables are now available for access within the `main.tf` file and can be set in the example module. The following example shows how to reference the `project_id` variable in `main.tf`.
+
+```hcl
+// name for your example module
+module "mysql-db" { 
+ ...
+ // variables required by the source module
+ random_instance_name = true
+ database_version     = "MYSQL_5_6"
+ 
+ // variable being set from the example module variables configuration. 
+ project_id           = var.project_id
+ ...
+}
+```
+
+```
+Note:
+Variables defined in the example module’s variables (normally variables.tf) configuration can be set 1) from 
+upstream modules calling the example module (e.g. fixtures) or 2) using environment variables by prefixing 
+the environment variable with “TF_VAR”. E.g. to set the project_id variable (above), setting the value in a 
+“TF_VAR_project_id” environment variable would automatically populate its value upon execution.
+```
+
+#### 3.1.1.2 Output variables for the test
+Upon successful execution of your example module, you will most likely need outputs for resources being provisioned to validate and assert in your test. This is done using outputs in Terraform. 
+1. In the `examples/mysql-public` directory create a file `outputs.tf`. The content for the file should be as follows:
+
+```hcl
+// The output value is set using the value attribute and is either computed in place or pulled from one of the recursive modules being called. In this case, it is being pulled from the mysql module that the example module is calling as its source.
+...
+output "mysql_user_pass" {
+ value       = module.mysql-db.generated_user_password
+ description = "The password for the default user. If not set, a random one will be generated and available in the generated_user_password output variable."
+}
+...
+```
+
+Complete code files for the example module can be found [here](https://github.com/terraform-google-modules/terraform-google-sql-db/tree/master/test/integration/mysql-public).
+
+### 3.1.2 Write an integration test
+After creating the example configuration, your example will automatically be tested and no further action is required.
+However, if you want to make custom assertions about what resources the blueprint will create, you should create an integration test in Go using the (testing)[https://pkg.go.dev/testing] package.
```

Comment:
We should add a sentence stating what the custom assertions could be - Custom assertions will mostly involve making an API call to GCL (via gcloud commands) to assert a resource is configured as expected.

Reply:
Added.